onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -167,20 +167,27 @@ def augment_graph(self):
 
         for tensor in tensors:
 
-            # Get tensor's shape 
-            dim = len(value_infos[tensor].type.tensor_type.shape.dim)
-            shape = (1,) if dim == 1 else list(1 for i in range(dim))
+            # Get tensor's shape.
+            # Don't keep dimension (keepdims=0) when reducing, only keep dimension if tensor contains any dim with value of 0.
+            dim = value_infos[tensor].type.tensor_type.shape.dim
+            keepdims = 0
+            shape = ()
+            for d in dim:
```

Comment:
why a for loop instead of a if check here?

Reply:
dim is an array.

I've modified the code on other branch
https://github.com/microsoft/onnxruntime/commit/a5e0c314fe47cf38b8120f9976b22ef1b8c18507